stores/tesco.js

```diff
@@ -0,0 +1,198 @@
+                } else {
+                    headers[pair[0]] = pair[1];
+                }
+            }
+        });
+        delete fetchOpts.body;
+
+        return headers.cookie.join("; ");
+    }
+    let parserOpts = {
+        lowerCaseTagName: true, // convert tag name to lower case (hurts performance heavily)
+        comment: false, // retrieve comments (hurts performance slightly)
+        blockTextElements: {
+            script: true,
+            noscript: false,
+            style: true,
+            pre: false,
+        },
+    };
+    let tescoItems = [];
+
+    const catRaw = await fetch("https://nakup.itesco.cz/groceries/cs-CZ/taxonomy");
+    let txt = await catRaw.text();
+    const categories = JSON.parse(txt);
+    const baseUrl = "https://nakup.itesco.cz/groceries/cs-CZ/shop";
+    let fetchOpts = {
+        headers: {
+            accept: "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7",
+            "accept-language": "cs,en;q=0.9,en-GB;q=0.8,en-US;q=0.7",
+            "cache-control": "no-cache",
+            pragma: "no-cache",
+            "sec-ch-ua": '"Not_A Brand";v="8", "Chromium";v="120", "Microsoft Edge";v="120"',
+            "sec-ch-ua-mobile": "?0",
+            "sec-ch-ua-platform": '"Windows"',
+            "sec-fetch-dest": "document",
+            "sec-fetch-mode": "navigate",
+            "sec-fetch-site": "same-origin",
+            "sec-fetch-user": "?1",
+            "upgrade-insecure-requests": "1",
+            referrerPolicy: "strict-origin-when-cross-origin",
+        },
+        body: null,
+    };
+
+    for (let i = 0; i < categories.length; i++) {
+        let main = categories[i];
+        const cat = main.name;
+        let childUrl = main.children[0].url;
+        let page = 1,
+            pagination;
+
+        do {
+            // https://nakup.itesco.cz/groceries/cs-CZ/shop/ovoce-a-zelenina/all (?page=1...)
+            const Url = `${baseUrl}${childUrl}?page=${page}&count=48`;
+            console.log(`${tescoItems.length} - ${i + 1}. z ${categories.length} ${Url}`);
+            if (!tescoItems.length) {
+                await firstGet(fetchOpts, Url);
+                fetchOpts.headers.cookie = headers.cookie.join("; ");
+                fetchOpts.method = "GET";
+            }
+
+            headers.cookie = [];
+            await fetch(Url, fetchOpts).then((response) => {
+                res = response;
+                for (const pair of response.headers) {
+                    if (pair[0] == "content-length") continue;
+                    if (pair[0] == "set-cookie") {
+                        headers.cookie.push(pair[1].split(";")[0]);
+                    } else {
+                        headers[pair[0]] = pair[1];
+                    }
+                }
+            });
+            // fetchOpts.headers.cookie = headers.cookie.join('; ');
```

Comment:
Zpočátku se mi po přesunu do firstGet stalo, že zas přestal načítat, ale asi je to zbytečné.